matrix/Vector.hpp

```diff
@@ -71,6 +71,11 @@ class Vector : public Matrix<Type, M, 1>
         return Type(sqrt(a.dot(a)));
     }
 
+    Type norm_squared() const {
+        const Vector &a(*this);
+        return Type(a.dot(a));
```

Comment:
Is it actually necessary to cast it?

Reply:
No, you're right, I forgot to get rid of it. It was only necessary for sqrt() which can return double...